### DISCUSSION TITLE
coding

### DISCUSSION THREAD

**Discussion by CalvinKoung** [bitcraze/discussions#1193] [Q&A]
Hi, I'm want to modify the code with lps tag algorithm and I downloaded the lps -node-firmware-master repositories in my laptop. It working good and no error in my mac. But when I put the all files open in my windows computer that strange thing happened. I recieved the error says cannot open source stdbool.h. How can I do ? thanks 

![2024-02-21 151826](https://github.com/bitcraze/discussions/assets/118436188/3eb6ab97-383b-4626-819e-28282fef5fb0)

**Comment by ataffanel** [comment]
We usually do not support firmware development on windows directly, on Windows you should run in WSL (linux) or a VM. We have tried to support development on pure windows a long time ago but while it might be possible, it was too much work to maintain for us (we are all on Linux/WSL).

**Reply by CalvinKoung** [reply]
ok. I got it
